### ISSUE TITLE
-[NSBundle yx_appShortVersion]: unrecognized selector sent to instance 0x6000011825d0 in [YMMYandexMetrica activateWithConfiguration:config]

### ISSUE THREAD

**Issue by ibovkush** [yandexmobile/metrica-plugin-cordova#21]
I'm using ionic 5 capacitor application. When I added the Yandex app metrica plugin to app, the iOS application start to crash on start. After investigation I found that the problem is in the string 182 in YMMAppMetricaPlugin.m file: 

`[YMMYandexMetrica activateWithConfiguration:config]`

the error is -[NSBundle yx_appShortVersion]: unrecognized selector sent to instance 0x6000011825d0 

I've tried to use only one value "API key (для использования в SDK)" 
I'm not sure how to check the problem. I even can't find the version on YMM framwork used in the plugin. 

Any ideas?

**Comment by Vlodzimej** [comment]
I have the same problem

**Comment by vovka-s** [comment]
Same problem.
ionic 4 + capacitor 2.4.2 + yandex-appmetrica-plugin-cordova 1.0.0
I get that error on any attempt to call reportEvent function.

**Comment by vovka-s** [comment]
Господа разработчики, хотя бы намекните, имеет ли смысл пытаться найти причину крешей?
Или для капаситора надо либу полностью переписывать?

На данный момент ситуация такая:
- связка ionic 4 + capacitor 2.4.2 + yandex-appmetrica-plugin-cordova 1.0.0
- работает на андроиде
- не работает не деске (`window.appMetrica = undefined`) (но само веб приложение работает нормально)
- крешится на ios при активации метрики
![креш из-за метрики](https://user-images.githubusercontent.com/9987233/104201094-6c7e8780-543a-11eb-808d-7e323ec3d682.png)

Пробовал
- по-разному изменять конфиг - ошибка не уходит
- подменять ios sdk на версии 3.14.0 и 3.9.4 - тоже возникают креши, но с другой ошибкой. Вероятно, что-то поменялось в АПИ
- даунгрейдить сам плагин - ошибка не уходит

В какую сторону копать, что ещё можно проверить/изменить?

**Comment by NesterovichAlexey** [comment]
Привет!
@vovka-s, можешь проверить стоит ли флаг линковщика `-ObjС` в `OTHER_LDFLAGS`?

**Comment by vovka-s** [comment]
> Привет!
> @vovka-s, можешь проверить стоит ли флаг линковщика `-ObjС` в `OTHER_LDFLAGS`?

Привет!
Проверял ранее и сейчас ещё раз посмотрел - этот флаг есть

**Comment by kamilsmtv** [comment]
Господа, есть успехи в данном вопросе? Точно такая же проблема

**Comment by rusekr** [comment]
+1 такая же проблема

**Comment by kamilsmtv** [comment]
> +1 такая же проблема

Я разрабатываю гибридное мобильное приложение на quasar с capacitor. Я смог решить проблему. В файле `src-capacitor/capacitor.config.json` дописал следующее 
```
...
"ios": {
    "cordovaLinkerFlags": [
      "-ObjC"
    ]
  }
...
```

**Comment by vaban-ru** [comment]
@vovka-s нашли в итоге решение этой проблемы?

**Comment by ghost** [comment]
Добрый день!Ваше письмо получено и зарегистрировано под уникальным номером № 14178384. Ответ обязательно будет предоставлен, пожалуйста, ожидайте.Просим Вас не отправлять повторное письмо по данному вопросу - оно будет считаться новой заявкой и может быть обработано позднее.Помощь сервиса: https://yandex.ru/support/direct/?from=email--С уважением,Отдел клиентского сервисател.: 8 800 234-24-80 (звонок из регионов России бесплатный)тел.: +7 495 739-37-77&nbsp30.09.2021, 12:43, "Vladimir Banduristov" ***@***.***>:
@vovka-s нашли в итоге решение этой проблемы?

**Comment by vovka-s** [comment]
> @vovka-s нашли в итоге решение этой проблемы?
Да. Помогло решение, предложенное в [этом](https://github.com/yandexmobile/metrica-plugin-cordova/issues/21#issuecomment-775738234) комменте

**Comment by vaban-ru** [comment]
> > @vovka-s нашли в итоге решение этой проблемы?
> > Да. Помогло решение, предложенное в [этом](https://github.com/yandexmobile/metrica-plugin-cordova/issues/21#issuecomment-775738234) комменте

Мне почему-то не помогает, а можешь поделиться списоком прилинкованных библиотек?

**Comment by vovka-s** [comment]
Сходу нет, этот код не пошёл в прод. Попробую поискать

**Comment by vaban-ru** [comment]
> Сходу нет, этот код не пошёл в прод. Попробую поискать

Буду благодарен, а то мне как раз в прод нужно лить уже, а из-за метрики этой всё крашится, я уже и конфиг дописывал, и руками добавлял в икскоде, один фиг не пашет, а на андроиде всё норм
